protocols/identify/src/behaviour.rs

```diff
@@ -63,18 +61,10 @@ pub struct Behaviour {
 }
 
 /// A pending reply to an inbound identification request.
-enum Reply {
-    /// The reply is queued for sending.
-    Queued {
-        peer: PeerId,
-        io: ReplySubstream<NegotiatedSubstream>,
-        observed: Multiaddr,
-    },
-    /// The reply is being sent.
-    Sending {
-        peer: PeerId,
-        io: Pin<Box<dyn Future<Output = Result<(), UpgradeError>> + Send>>,
-    },
+struct Request {
+    peer: PeerId,
+    io: ReplySubstream<NegotiatedSubstream>,
+    observed: Multiaddr,
```

Comment:
Same here, a `Handler` can learn this via `IntoConnectionHandler` and it won't change[^1] across its lifetime.

[^1]: `AddressChange` is not emitted at the moment but once it will, it is directly given to the handler.

Reply:
but we only get it on `connection_established` which seems to be called after `new_handler`.

Reply:
You have to implement `IntoConnectionHandler` additionally on a struct that you return from `new_handler`, said struct can then capture the `ConnectedPoint` and initialise the `ConnectionHandler` with it.

Reply:
thanks Thomas, updated!